File: medium/reverse_integer.go
package medium

func ReverseInteger(x int) int {
	num := int64(0)

	for x != 0 {
		num = num*10 + int64(x%10)
		x /= 10
	}
	if num > 1<<31-1 || num < -1<<31 {
		return 0
	}

	return int(num)
}

File: src/main/java/medium/ReverseInteger.java
package medium;
public class ReverseInteger {
		public int solve(int x) {
		long result = 0;
		while(x != 0) {
			result = result * 10 + x%10;
			x /= 10;
		}
		if(result > 0x7fffffff || result < Integer.MIN_VALUE) return 0;
		
	return (int) result;
	}
}